### ISSUE TITLE
Handling prop variables in tests

### ISSUE THREAD

**Issue by louisscruz** [cvazac/eslint-plugin-react-perf#26]
On recently upgrading, we found that our Jest/Enzyme tests have a bunch of new ESLint errors. Tests like:

```
describe('some component', () => {
  it('does the correct thing when color is red', () => {
    const color = 'red';
    const wrapper = shallow(<SomeComponent color={color} />);
    // ...
  });
});
```

Need to be re-written as:

```
const color = 'red';

describe('some component', () => {
  it('does the correct thing when color is red', () => {
    const wrapper = shallow(<SomeComponent color={color} />);
    // ...
  });
});
```

This is pretty far from ideal. I don't think that we will want to hoist all of test prop variables. It would be very difficult to read/reason about large test suites if we obey this rule there.

We're going to disable this plugin for our tests for now, but it seems like this might be something that should be handled out-of-the-box by the plugin in some way.

* Is there a way for `findRelevantNodes` to determine that the variable is declared in a non-React-component function? I'm not sure this is possible, but maybe there's a way.
* Could we maybe use some kind of callback exclusion list in the vein of https://github.com/cvazac/eslint-plugin-react-perf/issues/19?
* Should we be dealing with this in some other way? Or is it just preferred to turn the rules off for test files?

**Comment by cvazac** [comment]
You are right, the plugin is naive in that it validates JSX wherever it finds it, where it should only care about JSX that is returned by `render` methods. 

As a workaround, you could disable eslint (altogether or by rule) on a file by file basis like [here](https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments). I admit this workaround sucks. :(

A possible solution would be to pass the plugin an exclude pattern, something like: `"**/*.spec.js",`. Would that work for you?

**Comment by louisscruz** [comment]
I’m not sure that’d help much since that sort of glob opt-out behavior can already be achieved with ESLint configuration (https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files).

Would it be possible to scope this plugin, whether through some configuration setting or not, to only run for JSX that is returned from functions or JSX that is returned from class component render methods? I could see why it’s not possible. Any number of hacks could be made to sidestep an AST check for those things. But I’ve seen other prominent ESLint/Babel parsers that allow for a certain degree of incompleteness.

If this doesn’t seem feasible, feel free to close.

**Comment by cvazac** [comment]
I propose using config to specify an exclude list of function names. If the JSX is seen to be an argument to a method on the excludes list, no warning/error.

Would that work in all of your cases?
